lib/WeBWorK/PG/Translator.pm

```diff
@@ -1198,50 +1202,172 @@ sub process_answers{
+				                    Unrecognized evaluator type |". ref($rf_fun). "|");
+				$skip_evaluation=1;
 		}
-		$temp_ans  = $ans;
-		$temp_ans = '' unless defined($temp_ans); #make sure that answer is always defined
-		                                          # in case the answer evaluator forgets to check
-		$self->{safe}->share('$rf_fun','$temp_ans');
- 	    
+		if (not defined($new_temp_ans) ) {
+				$PG->warning_message( "No answer blank provided for answer evaluator $ans_name ");
+				$skip_evaluation=1;
+		}
+		$PG->debug_message("Answers associated with $ans_name are $new_temp_ans ref=". ref($new_temp_ans) ) if defined $new_temp_ans and $local_debug;
+       #FIXME -- this is a hack for handling check boxes. 
+        if (ref( $new_temp_ans) =~/HASH/i) {
+        	my @tmp = ();
+        	foreach my $key (sort keys %$new_temp_ans) {
+        		push @tmp, $key if $new_temp_ans->{$key} eq 'CHECKED';
+        	}
+        	$new_temp_ans = join("\0",@tmp);
+        	$PG->debug_message("Hash answer associated with $ans_name is $new_temp_ans. ref=". ref($new_temp_ans)."<br/>".pretty_print($new_temp_ans) ) if $local_debug;
+        }
+        #FIXME -- hack to allow answers such as <4,6,7>  -- < and > have been escaped.
+        unless( $skip_evaluation) { 
+			$new_temp_ans =~ s/\&lt\;/</g; # <
+			$new_temp_ans =~ s/\&gt\;/>/g; # >
+			$new_temp_ans =~ s/\&\#91\;/\[/g; # <
+			$new_temp_ans =~ s/\&\#42\;/\*/g; # *
+			$new_temp_ans =~ s/\&\#36\;/\$/g;  # $
+			$new_temp_ans =~ s/\&\#39\;/'/g;  # '
+			$new_temp_ans =~ s/\&\#95\;/\_/g;  # _
+        }
+		$self->{safe}->share('$new_rf_fun','$new_temp_ans');
+
+
+ 	     	    
  	    # clear %errorTable for each problem
  	    %errorTable = (); # is the error table being used? perhaps by math objects?
  	    
-		my $rh_ans_evaluation_result;
-		if (ref($rf_fun) eq 'CODE' ) {
-			$rh_ans_evaluation_result = $self->{safe} ->reval( '&{ $rf_fun }($temp_ans, ans_label => \''.$ans_name.'\')' ) ;
-			warn "Error in Translator.pm::process_answers: Answer $ans_name: |$temp_ans|\n $@\n" if $@;
-		} elsif (ref($rf_fun) =~ /AnswerEvaluator/)   {
-			$rh_ans_evaluation_result = $self->{safe} ->reval('$rf_fun->evaluate($temp_ans, ans_label => \''.$ans_name.'\')');
+		my ($rh_ans_evaluation_result, $new_rh_ans_evaluation_result);
+		
+		if (ref($new_rf_fun) eq 'CODE' ) {
+#			$rh_ans_evaluation_result = $self->{safe} ->reval( '&{ $rf_fun }($temp_ans, ans_label => \''.$ans_name.'\')' ) ;
+#			warn "Error in Translator.pm::process_answers: Answer $ans_name: |$temp_ans|\n $@\n" if $@;
+			$PG->warning_message( "Can no longer used CODE objects directly as answer evaluators.  Use AnswerEvaluator");
+		} elsif (not $skip_evaluation )   {
+# 			$rh_ans_evaluation_result = $self->{safe} ->reval('$rf_fun->evaluate($temp_ans, ans_label => \''.$ans_name.'\')');
+# 			$@ = $errorTable{$@} if $@ && defined($errorTable{$@});  #Are we redefining error messages here?
```

Comment:
I gather that this comment about errorTable is in response to the comment/question I inserted in the code and that you are not suggesting any changes here.  I will restore the original comment since that might help in the future.